lib/provisioning/Provisioner.js

```diff
@@ -91,62 +159,195 @@ Provisioner.prototype.isProvisionRequest = function(req) {
+    function*(roomId, userId, status, skey) {
+        let intent = this._ircBridge.getAppServiceBridge().getIntent();
+        try {
+            yield intent.client.sendStateEvent(roomId, 'm.room.bridging', {
+                user_id: userId,
+                status: status // pending, success, failure
+            }, skey);
+        }
+        catch (err) {
+            console.error(err);
+            throw new Error(`Could not update m.room.bridging state in this room`);
+        }
+    }
+);
+
+Provisioner.prototype._authoriseProvisioning = Promise.coroutine(
+    function*(server, userId, ircChannel, roomId, opNick, key) {
+        let ircDomain = server.domain;
+
+        let existing = this._getRequest(server, opNick);
+        if (existing) {
+            let from = existing.userId;
+            throw new Error(`Bridging request already sent to `+
+                            `${opNick} on ${server.domain} from ${from}`);
+        }
+
+        // (Matrix) Check power level of user
+        log.info(`Check power level of user`);
+        let matrixClient = this._ircBridge.getAppServiceBridge().getClientFactory().getClientAs();
+
+        let powerState = null;
+        try {
+            powerState = yield matrixClient.getStateEvent(roomId, 'm.room.power_levels');
+        }
+        catch (err) {
+            log.error(`Error retrieving power levels (${err.data.error})`);
+        }
+
+        if (!powerState) {
+            throw new Error('Could not retrieve your power levels for the room');
+        }
+
+        let actualPower = 0;
+        if (powerState.users[userId] !== undefined) {
+            actualPower = powerState.users[userId];
+        }
+        else if (powerState.users_default !== undefined) {
+            actualPower = powerState.users_default;
+        }
+
+        let requiredPower = 50;
+        if (powerState.events["m.room.power_levels"] !== undefined) {
+            requiredPower = powerState.events["m.room.power_levels"]
+        }
+        else if (powerState.state_default !== undefined) {
+            requiredPower = powerState.state_default;
+        }
+
+        if (actualPower < requiredPower) {
+            throw new Error('User does not possess high enough power level');
+        }
+
+        // (IRC) Check that op's nick is actually op
+        log.info(`Check that op's nick is actually op`);
+        let botClient = yield this._ircBridge.getBotClient(server);
+        yield botClient.joinChannel(ircChannel, key);
+        log.info(`Bot joined channel ${ircChannel}`);
+        let info = yield botClient.getOperators(ircChannel);
+
+        if (info.nicks.indexOf(opNick) == -1) {
+            let knownOps = info.operatorNicks.join(', ');
+            throw new Error(`Provided user is not in channel ${ircChannel}. ` +
+                            `Known ops in this channel: ${knownOps}`);
+        }
+
+        if (info.operatorNicks.indexOf(opNick) == -1) {
```

Comment:
Nit: `===` should be the default. Only use `==` when you have a reason to do so.